src/status.c

```diff
@@ -261,12 +261,33 @@ int git_status_list_new(
 	git_status_list **out,
 	git_repository *repo,
 	const git_status_options *opts)
+{
+  git_tree *head = NULL;
+  int error = 0;
+  
+  /* if there is no HEAD, that's okay - we'll make an empty iterator */
+  if ((error = git_repository_head_tree(&head, repo)) < 0) {
+    if (error != GIT_ENOTFOUND && error != GIT_EUNBORNBRANCH)
+      goto done;
+    giterr_clear();
+  }
+  error = git_status_list_new_at(out, repo, opts, head);
+  
+done:
+  git_tree_free(head);
+  return error;
+}
+
+int git_status_list_new_at(
+  git_status_list **out,
+  git_repository *repo,
+  const git_status_options *opts,
+  git_tree *head)
```

Comment:
You have a memory management issue here, as you try to `git_tree_free(head)` in the end. You should just make `head` const and have the caller manage memory for you.